lib/OpenQA/Worker/Job.pm

```diff
@@ -401,77 +401,97 @@ sub _stop_step_5_upload {
 
     # upload logs and assets
     if ($reason ne 'quit' && $reason ne 'abort' && $reason ne 'api-failure') {
-        # upload ulogs
-        my @uploaded_logfiles = glob "$pooldir/ulogs/*";
-        for my $file (@uploaded_logfiles) {
-            next unless -f $file;
-
-            my %upload_parameter = (
-                file => {file => $file, filename => basename($file)},
-                ulog => 1,
-            );
-            if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
-                $reason = 'api-failure';
-                last;
-            }
-        }
-
-        # upload assets created by successfull jobs
-        if ($reason eq 'done' || $reason eq 'cancel') {
-            for my $dir (qw(private public)) {
-                my @assets        = glob "$pooldir/assets_$dir/*";
-                my $upload_result = 1;
 
-                for my $file (@assets) {
+        Mojo::IOLoop->subprocess(
+            sub {
+                # upload ulogs
+                my @uploaded_logfiles = glob "$pooldir/ulogs/*";
+                for my $file (@uploaded_logfiles) {
                     next unless -f $file;
 
                     my %upload_parameter = (
-                        file  => {file => $file, filename => basename($file)},
-                        asset => $dir,
+                        file => {file => $file, filename => basename($file)},
+                        ulog => 1,
                     );
-                    last unless ($upload_result = $self->_upload_log_file_or_asset(\%upload_parameter));
+                    if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
+                        $reason = 'api-failure';
+                        last;
+                    }
                 }
-                if (!$upload_result) {
-                    $reason = 'api-failure';
-                    last;
+
+                # upload assets created by successfull jobs
```

Comment:
s/successfull/successful/

Reply:
I believe you wrote that comment originally. 😉

Reply:
Wouldn't be the first time I spot my own typos in another PR :-)